File: internal/game/team.go
package game

import "github.com/google/uuid"

type ActorConfig struct {
	Class   *uuid.UUID              `json:"class"`
	Items   []uuid.UUID             `json:"items"`
	Name    string                  `json:"name"`
	Weapons map[uuid.UUID]uuid.UUID `json:"weapons"`
}

type TeamConfig struct {
	Actors []ActorConfig `json:"actors"`
	Name   string        `json:"name"`
}

File: web/src/lib/game/team.ts
import z from 'zod'

export const ActorConfigSchema = z.object({
  class: z.uuid().nullable(),
  items: z.uuid().array(),
  name: z.string(),
  weapon_l: z.uuid().nullable(),
  weapon_r: z.uuid().nullable(),
})

export const TeamConfigSchema = z.object({
  name: z.string(),
  actors: z.array(ActorConfigSchema),
})

export const TeamSchema = z.object({
  ID: z.uuid().nullable(),
  config: TeamConfigSchema,
})

export type ActorConfig = z.infer<typeof ActorConfigSchema>
export type TeamConfig = z.infer<typeof TeamConfigSchema>
export type Team = z.infer<typeof TeamSchema>
